src/qibo/models/circuit.py

```diff
@@ -630,8 +630,11 @@ def set_parameters(self, parameters):
                 params = [0.123, 0.456, 0.789, 0.321]
                 c.set_parameters(params)
         """
+        import tensorflow as tf
         if isinstance(parameters, (list, tuple, np.ndarray)):
             self._set_parameters_list(parameters, len(parameters))
+        elif isinstance(parameters, (tf.Tensor, tf.Variable)):
+            self._set_parameters_list(parameters, int(parameters.shape[0]))
```

Comment:
There was an issue with tests regarding this.
Probably because we were missing this from the master branch.
I don't know if this is the best solution.
Let me know what you think @stavros11.

Reply:
Perfect, I also didn't like my first solution for the same reason.